tests/test_flows.py

```diff
@@ -380,43 +394,69 @@ def test_channel_selection(self):
         self.check_log(self.repeater1, "agent_wlan0", "CHANNEL_PREFERENCE_QUERY_MESSAGE")
         self.check_log(self.repeater1, "agent_wlan2", "CHANNEL_PREFERENCE_QUERY_MESSAGE")
 
-        # Send empty channel selection request
         self.debug("Send empty channel selection request")
-        self.gateway_ucc.dev_send_1905(self.mac_repeater1,0x8006)
+        self.gateway_ucc.dev_send_1905(self.mac_repeater1, 0x8006, tlv(0x00, 0x0000, "{}"))
+        time.sleep(1)
 
         self.debug("Confirming empty channel selection request has been received on agent")
         self.check_log(self.repeater1, "agent_wlan0", "CHANNEL_SELECTION_REQUEST_MESSAGE")
         self.check_log(self.repeater1, "agent_wlan2", "CHANNEL_SELECTION_REQUEST_MESSAGE")
 
-        self.debug("Confirming OPERATING_CHANNEL_REPORT_MESSAGE message has been received on controller")
-        self.check_log(self.gateway, "controller", "OPERATING_CHANNEL_REPORT_MESSAGE")
+        self.check_log(self.gateway, "controller", "CHANNEL_SELECTION_RESPONSE_MESSAGE")
+
+        self.debug("Confirming OPERATING_CHANNEL_REPORT_MESSAGE message \
+            has been received on controller with mid")
+
+        ocrm_line = 0
+        result, ocrm_line, mid = self.parse_log(self.gateway, "controller",
+                                                r'.+OPERATING_CHANNEL_REPORT_MESSAGE,\smid=(\d+)',
+                                                ocrm_line)
+        if (result):
```

Comment:
So with my suggestion this would become something like:
```suggestion
        ocrm_line, mid_match = self.check_log(self.gateway, "controller",
                                                r'OPERATING_CHANNEL_REPORT_MESSAGE, mid=(\d+)')
        if mid_match:
            mid = mid_match.group(1)
```

Reply:
In the updated implementation for check_log we should exactly write esult, ocrm_line, mid = self.check_log, because it returns bool int string, if we skip result we will get bool int, skipping string.